### DISCUSSION TITLE
Lonink VRF controller works out of the box

### DISCUSSION THREAD

**Discussion by sll1001** [merdok/homebridge-miot#483] [Show and tell]
Happy to report that this plugin works out of the box with Lonink VRF controller. 
For those who like me has been looking for a working VRF controller to integrate into the HomeKit environment, this product which is supported by MiHome works well. It is the Lonink B19 which I bought on Taobao.
As shown  in the dump below, each internal AC is presented as a separate device. homebridge-miot recognises it as air conditioner and configure correctly.

NAME:     LONINK VRF Central Air Conditioner Controller
   ID:       5901xxxxx
   MAC:     xxxx
   IP:       172.16.88.82
   TOKEN:    xxxx
   MODEL:    qdhkl.gateway.lonink
   ---------
   NAME:     VRF Air Conditioner
   ID:       proxy.5901xxxxx.0101
   MODEL:    qdhkl.aircondition.ac
   ---------
   NAME:     VRF Air Conditioner 2
   ID:       proxy.5901xxxxx.0102
   MODEL:    qdhkl.aircondition.ac
   ---------
   NAME:     VRF Air Conditioner 3
   ID:       proxy.5901xxxxx.0103
   MODEL:    qdhkl.aircondition.ac
   ---------
   NAME:     VRF Air Conditioner 4
   ID:       proxy.5901xxxxx.0201
   MODEL:    qdhkl.aircondition.ac
   ---------
   NAME:     VRF Air Conditioner 5
   ID:       proxy.5901xxxxx.0202
   MODEL:    qdhkl.aircondition.ac
   --------
   NAME:     VRF Air Conditioner 6
   ID:       proxy.590173586.0203
   MODEL:    qdhkl.aircondition.ac
  --------

Here is a config example:
{
                    "name": "Living MFC Aircon",
                    "ip": "172.16.xxx.xxx",
                    "token": "xxxx",
                    "deviceId": "proxy.5901xxxxx.0201",
                    "model": "qdhkl.aircondition.ac",
                    "pollingInterval": 10,
                    "deepDebugLog": true,
                    "onlyMainService": true,
                    "buzzerControl": false,
                    "ledControl": false,
                    "childLockControl": false,
                    "modeControl": false,
                    "suppressAutoServiceCreation": [
                        "temperature",
                        "relativeHumidity",
                        "illumination"
                    ]
                }

**Comment by merdok** [comment]
Good to hear😃 Thanks for reporting!

**Comment by ChinaXing** [comment]
is this  support control fan level of the air-conditional ?

**Reply by sll1001** [reply]
The way I set it up, as shown in the config example in my previous posting, there is no fan control.
However, the miot spec has a fan control property so it is likely to be supported. Just have to change the config parameter a bit.
Worth a try.

**Reply by merdok** [reply]
If the device does have and it is not shown then you can use the propertyControl to expose that in your home app.

**Comment by merdok** [comment]
When the device supports it then probably yes.
